File: packages/sdk/java/packages/core/src/main/java/dev/edgebase/sdk/core/FieldOps.java
package dev.edgebase.sdk.core;

import java.util.HashMap;
import java.util.Map;

/** Field operation helpers for atomic updates. */
public final class FieldOps {
    private FieldOps() {}

    public static Map<String, Object> increment(Number value) {
        Map<String, Object> op = new HashMap<>();
        op.put("$op", "increment");
        op.put("value", value);
        return op;
    }

    public static Map<String, Object> deleteField() {
        Map<String, Object> op = new HashMap<>();
        op.put("$op", "deleteField");
        return op;
    }
}

File: packages/sdk/kotlin/core/src/commonMain/kotlin/dev/edgebase/sdk/core/FieldOps.kt
// EdgeBase Kotlin SDK — Field operation helpers.
//
// Atomic field operations: increment, deleteField.
// Format mirrors JS SDK (field-ops.ts) and server op-parser.ts ($op key).

package dev.edgebase.sdk.core

/**
 * Field operation helpers for atomic updates.
 *
 * Usage:
 * ```kotlin
 * docRef.update(mapOf("views" to FieldOps.increment(1)))
 * docRef.update(mapOf("temp" to FieldOps.deleteField()))
 * ```
 */
object FieldOps {
    /**
     * Atomically increment a numeric field.
     */
    fun increment(value: Number): Map<String, Any> = mapOf(
        "\$op" to "increment",
        "value" to value
    )

    /**
     * Mark a field for deletion.
     */
    fun deleteField(): Map<String, Any> = mapOf(
        "\$op" to "deleteField"
    )
}
